pandas/io/parsers/python_parser.py

```diff
@@ -555,20 +555,18 @@ def _infer_columns(
                 columns = self._handle_usecols(
                     columns, columns[0], num_original_columns
                 )
-            else:
-                if self.usecols is None or len(names) >= num_original_columns:
-                    columns = self._handle_usecols([names], names, num_original_columns)
-                    num_original_columns = len(names)
-                else:
-                    if not callable(self.usecols) and len(names) != len(self.usecols):
-                        raise ValueError(
-                            "Number of passed names did not match number of "
-                            "header fields in the file"
-                        )
-                    # Ignore output but set used columns.
-                    self._handle_usecols([names], names, ncols)
-                    columns = [names]
-                    num_original_columns = ncols
+            elif self.usecols is None or len(names) >= num_original_columns:
+                columns = self._handle_usecols([names], names, num_original_columns)
+                num_original_columns = len(names)
+                if not callable(self.usecols) and len(names) != len(self.usecols):
+                    raise ValueError(
+                        "Number of passed names did not match number of "
```

Comment:
does this still need to be in the else block?

Reply:
it was before changing to elif.

Reply:
How did you know error was on that specific line???